### PR TITLE
Add pagination to load branches

### PR DESCRIPTION
**Pull request by kuzaxak**
* Add pagination to loading of list branches

This patch is compatible with gitlab api v3

### CODE REVIEW

File: lib/Gitlab/Api/Repositories.php [hunk 1]
```diff
@@ -4,11 +4,15 @@ class Repositories extends AbstractApi
...
      * @param int $project_id
+     * @param array $parameters
      * @return mixed
      */
-    public function branches($project_id)
+    public function branches($project_id, array $parameters = [])
     {
-        return $this->get($this->getProjectPath($project_id, 'repository/branches'));
+        $resolver = $this->createOptionsResolver();
+        $resolver->setDefined('query');
```
Comment (fbourigault) [inline lib/Gitlab/Api/Repositories.php]: According to https://docs.gitlab.com/ee/api/branches.html#list-repository-branches, there is no `query` filter available for this branch. Could you remove it?

Comment (kuzaxak) [inline lib/Gitlab/Api/Repositories.php]: In api v4 when you try load list of branches you receive Header with pagination. Also pagination query works. 

Note: you can compare queries.
 
V3: 
http://take.ms/e4hJW

V4:
http://take.ms/NNhQK
http://take.ms/8AD67

Comment (kuzaxak) [inline lib/Gitlab/Api/Repositories.php]: I think it is issue related with API documentation.

**Comment by blankse** [comment]
@kuzaxak Can you also add pagination for tags?

**Comment by fbourigault** [comment]
Please, open an other PR for tags!

File: lib/Gitlab/Api/Repositories.php (hunk 1 above)
Comment (fbourigault) [reply lib/Gitlab/Api/Repositories.php]: As the filter is undocumented, I would not add it for now and try to learn more about why it's not documented.

Comment (kuzaxak) [reply lib/Gitlab/Api/Repositories.php]: Without this fix you can't load all branches.

Comment (fbourigault) [reply lib/Gitlab/Api/Repositories.php]: You can using the `Gitlab\ResultPager::fetchAll()`!

However, I'm not willing to add undocumented methods/parameters to this library. There is probably a reason about it not being documented. We can move forward with this PR by removing the `query` option, and add-it later if it was a Gitlab API documentation issue.

**Comment by tobiasetter** [comment]
It is already possible to use pagination for branches so I think we don't need a merge here.

The only thing we could add is a test for the pagination as I added for the project members.

**Comment by GrahamCampbell** [comment]
It looks like this has already been implemented.

**Pull request closed without merging**